test/specs/entityReferenceValues.js

```diff
@@ -0,0 +1,51 @@
+import { expect } from "chai";
+import {getRowCount, userLogin} from "../../helper/methods";
+import selectors from "./../../data/selectors.json";
+import expected from "./../../data/expected.json";
+import testData from "./../../data/testData.json";
+
+describe('USER LOGIN,CREATE AND SAVE NEW RECORD', () => {
+    before(() => {
+        browser.maximizeWindow();
+        browser.url('https://ref2.eteam.work');
+        userLogin('tester39@tester.test', 'xGY9wLIMcs');
+    })
+    it('01/RefVal-Should go to menu Reference Values', () => {
+        $(selectors.navigationBar.userProfile).click();
+        $(selectors.navigationBar.resetAll).click();
+        $(selectors.navigationBar.sideBar).click();
+        $(selectors.referenceValues.referenceValuesMenu).scrollIntoView();
+        $(selectors.referenceValues.referenceValuesMenu).click();
+        expect($(selectors.referenceValues.header3).getText())
+            .to.equal(expected.referenceValues.headerOfBoardRefVal);
+    });
+    it('02/RefVal-Should create new record', () => {
+        $(selectors.common.createRecordBtn).click();
+        expect($(selectors.referenceValues.recordForm).isDisplayed()).true;
+    });
+    it('03/LP-Should input data into new record', () => {
+        $(selectors.referenceValues.labelIF)
+            .setValue(testData.referenceValues.label);
+        $(selectors.referenceValues.filter1IF)
+            .setValue(testData.referenceValues.filter1);
+        $(selectors.referenceValues.filter2IF)
+            .setValue(testData.referenceValues.filter2);
+        expect($(selectors.referenceValues.labelIF)
+            .getValue() === expected.referenceValues.label);
+        expect($(selectors.referenceValues.filter1IF)
+            .getValue()===expected.referenceValues.filter1);
+        expect($(selectors.referenceValues.filter2IF)
+            .getValue()===expected.referenceValues.filter2);
+        expect($(selectors.referenceValues.saveBtn).isClickable());
+    });
+    it('04/LP-Should save new record', () => {
+        $(selectors.referenceValues.saveBtn).click();
+        expect(getRowCount()).to.equal(expected.referenceValues.one);
+    });
+    it('05/LP-Should verify the values are visible and valid', () => {
+        for (let i=0;i < $(selectors.referenceValues.listOfTable).length; i++) {
+            expect($(selectors.referenceValues.listOfTable)[i].getText())
+                .to.equal(expected.referenceValues.expectedValue[i]);
```

Comment:
this for loop doesn't work because $(selectors.referenceValues.listOfTable).length == undefined

so you can fix it here 
```suggestion
        for (let i=0; i < $$(selectors.referenceValues.listOfTable).length; i++) {
            expect($$(selectors.referenceValues.listOfTable)[i].getText())
                .to.equal(expected.referenceValues.expectedValue[i]);
```

or call method **checkRecordActualData(expectedData)**